src/net/mod.rs

```diff
@@ -113,8 +113,16 @@ where
     }
 
     /// Enable live data streaming.
-    pub fn enable_streaming(&mut self) -> BlockGenerator {
-        self.generator.take().unwrap()
+    ///
+    /// # Args
+    /// * `format` - A unique u8 code indicating the format of the data.
+    pub fn enable_streaming(
```

Comment:
This doesn't really enable streaming.
More `get_streaming_generator(format: impl Into<u8>)`?